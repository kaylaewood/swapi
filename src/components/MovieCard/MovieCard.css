.movie-card {
  background-color: #0B1E36;
  color: white;
  text-align: center;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  padding: 2vw;
  margin: 2vw;
  box-shadow: 15px 15px 15px #A6A6A6;
}

.movie-title{
  margin: 1vw;
}

.view-characters-button {
  font-size: 1.4vw;
  width: 20vw;
}
