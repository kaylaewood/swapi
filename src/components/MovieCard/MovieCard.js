import React from 'react';
import './MovieCard.css';

function MovieCard({ movie, switchPages }) {
  return (
    <section className='movie-card'>
      <h3>EPISODE {movie.episode_id}</h3>
      <h2 className='movie-title'>{movie.title.toUpperCase()}</h2>
      <h4>RELEASED IN {movie.release_date.split('-')[0]}</h4>
      <button
        className='view-characters-button'
        // onClick={() => switchPages('characters')}
      >
      VIEW CHARACTERS
      </button>
    </section>
  );
}

export default MovieCard;
