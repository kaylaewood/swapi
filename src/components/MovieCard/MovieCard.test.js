import React from 'react';
import ReactDOM from 'react-dom';
import MovieCard from './MovieCard';
import { shallow } from 'enzyme';

describe('MovieCard', () => {
  let movie = {
    title: 'Die Hard',
    release_date: '1988-07-15'
  }
  it('should match the snapshot', () => {
    const wrapper = shallow(<MovieCard
      key={1}
      movie={movie}
    />)
    expect(wrapper).toMatchSnapshot();
    })
})
