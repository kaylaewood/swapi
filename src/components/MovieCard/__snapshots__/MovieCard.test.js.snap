// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match the snapshot 1`] = `
<section
  className="movie-card"
>
  <h3>
    EPISODE 
  </h3>
  <h2
    className="movie-title"
  >
    DIE HARD
  </h2>
  <h4>
    RELEASED IN 
    1988
  </h4>
  <Link
    to="/movies/undefined"
  >
    <button
      className="view-characters-button"
      onClick={[Function]}
    >
      VIEW CHARACTERS
    </button>
  </Link>
</section>
`;
